java-checks-testkit/src/main/java/org/sonar/java/checks/verifier/internal/InternalCheckVerifier.java

```diff
@@ -153,13 +153,19 @@ public InternalCheckVerifier withQuickFixes() {
   @Override
   public InternalCheckVerifier withJavaVersion(int javaVersionAsInt) {
     requiresNull(javaVersion, "java version");
-    javaVersion = new JavaVersionImpl(javaVersionAsInt);
-    return this;
+    return withJavaVersion(javaVersionAsInt, false);
   }
   
   @Override
   public InternalCheckVerifier withJavaVersion(int javaVersionAsInt, boolean enablePreviewFeatures) {
     requiresNull(javaVersion, "java version");
+    if (enablePreviewFeatures && javaVersionAsInt < JavaVersionImpl.MAX_SUPPORTED) {
+      var message = String.format(
+        "Preview features can only be enabled for the latest supported Java version (%d)",
```

Comment:
`for the latest supported Java version (%d)` => `when the version >= latest supported Java version (>= %d)`